actioncable/lib/action_cable.rb

```diff
@@ -35,7 +37,15 @@ module ActionCable
       confirmation: 'confirm_subscription'.freeze,
       rejection: 'reject_subscription'.freeze
     },
-    default_mount_path: '/cable'.freeze
+    default_mount_path: '/cable'.freeze,
+    # It is important that the unsupported protocol be last in the list,
+    # as the Websocket Driver always chooses the first protocol that matches,
+    # with the intended result that if the client sends the unsupported
+    # protcol as an option, it will be the last chosen.
```

Comment:
Could manifest the order requirement by using `protocols.last` in code instead of an extra key. Other parts of the code might even want to stay oblivious to the fact that last protocol means "unsupported", and write it as "whelp, we're all out of protocols, no reason to attempt reconnect."